### ISSUE TITLE
Comparing nullable decimal property 1.1m and 1.10m in complex type collection resulting "not equal"

### ISSUE THREAD

**Issue by leGodinho** [GregFinzer/Compare-Net-Objects#187]
When comparing a collection of object that contains a property from type nullable decimal (decimal?) and the values differ only on precision right zero value like `1.1m` and `1.10m`, comparer results with a wrong `"AreEqual = false"` result and shows 2 diferences, this issue only occur when `"IgnoreCollectionOrder = true"`, even when `DecimalPrecision `is set` to 0.1m` or any other value;

```
Begin Differences (2 differences):
Types [TestClass,null], Item Expected[DecimalValue:1,1].Item != Actual[DecimalValue:1,1].Item, Values (Bcx.Importador.Tests.TestClass,(null))
Types [null,TestClass], Item Expected[DecimalValue:1,10].Item != Actual[DecimalValue:1,10].Item, Values ((null),Bcx.Importador.Tests.TestClass)
End Differences (Maximum of 100 differences shown).
```

- When setting `"IgnoreCollectionOrder = false"`,the result `AreEqual`  is true;
- When changing type from `decimal?` to `decimal`, the result `AreEqual`  is true;
- When comparing values with same precision like `1.1m` and `1.1m`, the result `AreEqual`  is true;
- When comparing ingle nullable decimals out of collection, the result `AreEqual`  is true;
- When comparing complex type that has a property nullable decimal that is not in a collection, the result `AreEqual`  is true;

Below a code that simules the issue:

```
using KellermanSoftware.CompareNetObjects;
using NUnit.Framework;

namespace Bcx.Importador.Tests
{
    public class TestClass
    {
        public decimal? DecimalValue { get; set; }

        public static CompareLogic GetComparer()
        {
            return new CompareLogic
            {
                Config = new ComparisonConfig
                {
                    MaxDifferences = 100,
                    IgnoreCollectionOrder = true,
                    DecimalPrecision = 0.1m
                }
            };
        }
    }

    public class Tests
    {
        [SetUp]
        public void Setup()
        {
        }

        [Test]
        public void Test1()
        {
            var test1 = new[] { new TestClass { DecimalValue = 1.1m } };
            var test2 = new[] { new TestClass { DecimalValue = 1.10m } };

            var compareResult = TestClass.GetComparer().Compare(test2, test1);

            Assert.IsTrue(compareResult.AreEqual);
        }
    }
}
```

More dependency details:

- .NET Core 3.0
- CompareNETObjects/4.64.0
- NUnit/3.12.0
- NUnit3TestAdapter/3.15.1
- Microsoft.NET.Test.Sdk/16.4.0

**Comment by GregFinzer** [comment]
It will be a while until I can take a look at this

**Comment by leGodinho** [comment]
I was debugging here and found the point of issue, in KellermanSoftware.CompareNetObjects.IgnoreOrderTypes.IgnoreOrderLogic method GetMatchIndex() line 223.

`string formatString = $"{{0}}:{{1{(info.PropertyType.Name == "Decimal" ? ":N" : string.Empty)}}},";`

 There is a ternary operator comparing type PropertyType.Name with the constant "Decimal" but in this case, the name is getting "Nullable`1" value, I belive that changing the code to use PropertyType.FullName and comparing with string.Contains() should solve the problem.

`string formatString = $"{{0}}:{{1{(info.PropertyType.FullName.Contains("System.Decimal") ? ":N" : string.Empty)}}},";`

I will submit a pull request and if you spend a little time just to review than I think we can close this issue early. By the way I will change my property type to use decimal instead of nullable decimal as a workaround.

**Comment by GregFinzer** [comment]
Please do something similar to what we are doing for TypeHelper.IsDecimal
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types

**Comment by GregFinzer** [comment]
Released in v4.65
